### PR TITLE
dask: Basic daskified reading and writing

### PR DESCRIPTION
**Pull request by davidhassell**
* This PR requires https://github.com/NCAS-CMS/cfdm/pull/196 to be merged into the (master of) `cfdm` being used.
* Only tests `setup_create_field.py` and `test_read_write.py` have been tested. The latter has some skips relating to `cf.aggregate`, which will be fixed elsewhere.
* Running these test on my local machine "hasn't seg faulted for a while". _edit - Oh, it just did 2 out of the last 3 times I ran test_read_write.py_

### CODE REVIEW

**Comment by davidhassell** [comment]
I'm fairly sure that the seg faults are something to do with file open/close commands on chunks

File: cf/read_write/netcdf/netcdfread.py [hunk 1]
```diff
@@ -412,26 +326,76 @@ def _customize_read_vars(self):
...
+            # --------------------------------------------------------
+            # netCDF4 does not auto-mask VLEN variable, so do it here.
+            # --------------------------------------------------------
+            array = np.ma.where(array == b"", np.ma.masked, array)
+
+        elif not string_type and kind == "O":
+            # --------------------------------------------------------
+            # A netCDF string type N-d (N>=1) variable comes out as a
+            # numpy object array, so convert it to numpy string array.
+            # --------------------------------------------------------
```
Comment (sadielbartholomew) [inline cf/read_write/netcdf/netcdfread.py:398]: Not that it matters or we win any awards for it (though the good commenting is noted, as it often is), but there's a satisfying alignment across all of the comments here :smiley:

**Review by sadielbartholomew — APPROVED** [review]
Looks good, and the relevant test modules pass for me locally. The full test suite did crash my laptop, but that is unlikely to be related and we can worry about that later!

Please merge.

File: cf/read_write/netcdf/netcdfread.py (hunk 1 above)
Comment (sadielbartholomew) [reply cf/read_write/netcdf/netcdfread.py:398]: On second look, partly because one comment is the same as another, but still...

**Pull request merged**